From a52aa624e69fa618aa0d29c2a147a7d472751de1 Mon Sep 17 00:00:00 2001
From: Jake Petroules <jake.petroules@qt.io>
Date: Wed, 17 Aug 2016 12:59:37 -0700
Subject: [PATCH] Fix instance method not found warning

NSView/UIView were only forward declared here which led to warnings when
calling methods on instances of them.

Change-Id: Ic2b391bb0ed8d45306dc16e2a807ce7fcae5015e
Reviewed-by: Gabriel de Dietrich <gabriel.dedietrich@qt.io>
---
 src/plugins/avfoundation/mediaplayer/avfvideowidget.mm | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/plugins/avfoundation/mediaplayer/avfvideowidget.mm b/src/plugins/avfoundation/mediaplayer/avfvideowidget.mm
index f00146d1b..22cd41e69 100644
--- src/plugins/avfoundation/mediaplayer/avfvideowidget.mm.orig
+++ src/plugins/avfoundation/mediaplayer/avfvideowidget.mm
@@ -39,6 +39,12 @@
 #include <QtGui/QPaintEvent>
 #include <QtGui/QPainter>
 
+#if defined(Q_OS_OSX)
+#import <AppKit/AppKit.h>
+#else
+#import <UIKit/UIKit.h>
+#endif
+
 QT_USE_NAMESPACE
 
 AVFVideoWidget::AVFVideoWidget(QWidget *parent)
-- 
GitLab

